odxtools/dtcdop.py

```diff
@@ -201,14 +248,52 @@ def _resolve_odxlinks(self, odxlinks: OdxLinkDatabase) -> None:
                 dtc = odxlinks.resolve(dtc_proxy, DiagnosticTroubleCode)
                 self._dtcs.append(dtc)
 
-        linked_dtc_dops = [odxlinks.resolve(x, DtcDop) for x in self.linked_dtc_dop_refs]
-        self._linked_dtc_dops = NamedItemList(linked_dtc_dops)
+        for linked_dtc_dop in self.linked_dtc_dops_raw:
+            linked_dtc_dop._resolve_odxlinks(odxlinks)
 
     def _resolve_snrefs(self, context: SnRefContext) -> None:
+        # hack to avoid initializing the DtcDop object multiple
+        # times. This is required, because the linked DTC DOP feature
+        # requires the "parent" DTC DOPs to be fully initialized but
+        # the standard does not define a formal ordering of DTC DOPs.
+        if hasattr(self, "_linked_dtc_dops"):
```

Comment:
hasattr have performance impact and mypy can't check it, better to use a bool attribute set it to False in init, and True here

Reply:
IMO the performance impact is not relevevant here: this function is only called once per `DtcDop` when loading a database, IOW typically a few dozen times. I guess I'll implement the solution which you've proposed tomorrow in order to see if I like it better than the current hack. stay tuned...